server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java

```diff
@@ -688,22 +693,53 @@ private BoolQueryBuilder createBoolFilter(IssueQuery query) {
           .subAggregation(AggregationBuilders.count(projectUuid + "_count").field(IssueIndexDefinition.FIELD_ISSUE_KEY))
-          .subAggregation(AggregationBuilders.max(projectUuid + "_maxFuncCreatedAt").field(IssueIndexDefinition.FIELD_ISSUE_FUNC_CREATED_AT))
-        );
+          .subAggregation(AggregationBuilders.max(projectUuid + "_maxFuncCreatedAt").field(IssueIndexDefinition.FIELD_ISSUE_FUNC_CREATED_AT)));
     });
     SearchResponse response = request.get();
     return response.getAggregations().asList().stream()
       .map(x -> (InternalFilter) x)
       .flatMap(projectBucket -> {
-      long count = ((InternalValueCount) projectBucket.getAggregations().get(projectBucket.getName() + "_count")).getValue();
-      if (count < 1L) {
-        return Stream.empty();
-      }
-      long lastIssueDate = (long) ((InternalMax) projectBucket.getAggregations().get(projectBucket.getName() + "_maxFuncCreatedAt")).getValue();
-      return Stream.of(new ProjectStatistics(projectBucket.getName(), count, lastIssueDate));
-    }).collect(MoreCollectors.toList(projectUuids.size()));
+        long count = ((InternalValueCount) projectBucket.getAggregations().get(projectBucket.getName() + "_count")).getValue();
+        if (count < 1L) {
+          return Stream.empty();
+        }
+        long lastIssueDate = (long) ((InternalMax) projectBucket.getAggregations().get(projectBucket.getName() + "_maxFuncCreatedAt")).getValue();
+        return Stream.of(new ProjectStatistics(projectBucket.getName(), count, lastIssueDate));
+      }).collect(MoreCollectors.toList(projectUuids.size()));
   }
+
+  public List<BranchStatistics> searchBranchStatistics(List<String> branchUuids) {
+    if (branchUuids.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    SearchRequestBuilder request = client.prepareSearch(IssueIndexDefinition.INDEX_TYPE_ISSUE)
+      .setQuery(
+        boolQuery()
+          .must(termsQuery(IssueIndexDefinition.FIELD_ISSUE_BRANCH_UUID, branchUuids))
+          .must(termQuery(IssueIndexDefinition.FIELD_ISSUE_IS_MAIN_BRANCH, Boolean.toString(false))))
+      .setSize(0);
+    IntStream.range(0, branchUuids.size()).forEach(i -> {
+      String branchUuid = branchUuids.get(i);
+      request
+        .addAggregation(AggregationBuilders
+          .filter(branchUuid, boolQuery()
+            .must(termQuery(IssueIndexDefinition.FIELD_ISSUE_BRANCH_UUID, branchUuid))
+            .must(termQuery(IssueIndexDefinition.FIELD_ISSUE_IS_MAIN_BRANCH, Boolean.toString(false))))
+          .subAggregation(
+            AggregationBuilders.terms("types").field(IssueIndexDefinition.FIELD_ISSUE_TYPE)));
+    });
+    SearchResponse response = request.get();
+    return response.getAggregations().asList().stream()
+      .map(x -> (InternalFilter) x)
+      .flatMap(bucket -> {
+        Map<String, Long> typesCount = ((StringTerms) bucket.getAggregations().get("types")).getBuckets().stream()
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "StringTerms". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1905)